docs/data/data-grid/filtering/index.md

```diff
@@ -127,6 +127,26 @@ In the example below, the _rating_ column can not be filtered.
 
 {{"demo": "DisableFilteringGridSomeColumns.js", "bg": "inline", "defaultCodeOpen": false}}
 
+### Filter non-filterable columns programmatically
```

Comment:
Update in context of https://github.com/mui/mui-x/pull/11512#discussion_r1444724567
Sidenote: I've kept internal variable naming to have the keyword `readOnly` for better understanding. The alternative `disabled` may also convey `not visible at all`, so to me `readOnly` is the more appropriate term.
Let me know if there's a disagreement.